src/monitor/node_active_protocol.c

```diff
@@ -965,7 +989,153 @@ stop_maintenance(PG_FUNCTION_ARGS)
 					  currentNode->nodePort,
 					  currentNode->pgsrSyncState,
 					  currentNode->reportedLSN,
+					  currentNode->candidatePriority,
+					  currentNode->replicationQuorum,
 					  message);
 
 	PG_RETURN_BOOL(true);
 }
+
+
+/*
+ * set_node_candidate_priority sets node candidate priority property
+ */
+Datum
+set_node_candidate_priority(PG_FUNCTION_ARGS)
+{
+	int32 nodeId = PG_GETARG_INT32(0);
+	text *nodeNameText = PG_GETARG_TEXT_P(1);
+	char *nodeName = text_to_cstring(nodeNameText);
+	int32 nodePort = PG_GETARG_INT32(2);
+	int candidatePriority = PG_GETARG_INT32(3);
+	AutoFailoverNode *currentNode = NULL;
+	bool nodeUpdated = false;
+
+	ereport(LOG, (errmsg("updating candidate_priority for node %d, candidate_priority=%d",
+						  nodeId, candidatePriority)));
+
+	checkPgAutoFailoverVersion();
+
+	currentNode = GetAutoFailoverNodeWithId(nodeId, nodeName, nodePort);
+
+	if (currentNode == NULL)
+	{
+		ereport(ERROR, (errmsg("node %d is not registered", nodeId)));
+	}
+
+	LockFormation(currentNode->formationId, ShareLock);
+	LockNodeGroup(currentNode->formationId, currentNode->groupId, ExclusiveLock);
+
+	if (candidatePriority < 0 || candidatePriority > 100)
+	{
+		ereport(ERROR, (ERRCODE_INVALID_PARAMETER_VALUE,
+						errmsg("invalid value for candidate_priority \"%d\" "
+							   "expected an integer value between 0 and 100", candidatePriority)));
+	}
+
+
+	if (candidatePriority != currentNode->candidatePriority)
+	{
+		currentNode->candidatePriority = candidatePriority;
+		nodeUpdated = true;
+	}
+
+	if (nodeUpdated)
+	{
+		char message[BUFSIZE];
+
+		ReportAutoFailoverNodeReplicationSetting(currentNode->nodeId,
+				currentNode->nodeName,
+				currentNode->nodePort,
+				currentNode->candidatePriority,
+				currentNode->replicationQuorum);
+
+		LogAndNotifyMessage(
+			message, BUFSIZE,
+			"Updating candidatePriority.");
+
+		NotifyStateChange(currentNode->reportedState,
+						  currentNode->goalState,
+						  currentNode->formationId,
+						  currentNode->groupId,
+						  currentNode->nodeId,
+						  currentNode->nodeName,
+						  currentNode->nodePort,
+						  currentNode->pgsrSyncState,
+						  currentNode->reportedLSN,
+						  currentNode->candidatePriority,
+						  currentNode->replicationQuorum,
+						  message);
+
+	}
+
+	PG_RETURN_BOOL(nodeUpdated);
```

Comment:
And so if the user sets the same value again, then we report a failure to update?